### PR TITLE
Mr miles ca cert

### LINKED ISSUE #901: Add certificate options to vault_consul_secret_backend
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version

Terraform v0.12.29
+ provider.vault v2.15.0

### Affected Resource(s)

- vault_consul_secret_backend

### Terraform Configuration Files
```
resource "vault_consul_secret_backend" "consul" {
  path        = "consul"
  description = "Manage the Consul backend"

  scheme = "https"
  address = "consul.network.com"
  token   = var.vault_token
  ca_cert = "xxx" <-- missing

}
```

### Expected Behavior

There should be a parameter to set the ca_cert for the backend.  It can be set vai the cli using the /config/access endpoint.

### Actual Behavior
There isn't a setting to set the certificate for the backend.  Worse, if you do set the certificate from the command line then it is reset next time terraform runs causing breakage.

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
We have several distinct consul servers with different certificates so I specifically want to set it per-secret-engine rather than globally.

### PR DESCRIPTION
**Pull request by mr-miles**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #901 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
Extend consul secret engine definition to cover all vault parameters
```

Output from acceptance testing:

(running on windows [shudder] so here's the circleci output)

```

https://circleci.com/api/v1.1/project/github/terraform-providers/terraform-provider-vault/1264/output/104/0?file=true&allocation-id=5fb456f218e33a661f94ba9f-0-build%2F3179242F

```

### CODE REVIEW

**Comment by mr-miles** [comment]
The check failure is because the PR description was longer than jira's max issue size.  It is not related to the PR code.

File: vault/resource_consul_secret_backend.go [hunk 1]
```diff
@@ -166,6 +196,13 @@ func consulSecretBackendRead(d *schema.ResourceData, meta interface{}) error {
 	// So... if it drifts, it drift.
 	d.Set("address", secret.Data["address"].(string))
 	d.Set("scheme", secret.Data["scheme"].(string))
+	
+	val, ok := secret.Data["ca_cert"]
+	if ok && val != nil {
+	    d.Set("ca_cert", val.(string))
+	} else {
+	    d.Set("ca_cert", "")
+	}
```
[the remark this answers is not in the document]
Comment (mr-miles) [reply vault/resource_consul_secret_backend.go]: Thanks - I hadn't spotted that it thought it was constantly changing.  I was also confused why it wasn't returning the ca_cert data (had it down to "operator error"!) so thanks also for that vault source link

**Review by tvoran — COMMENTED** [review]
The other thing I'd suggest is to test those new parameters a bit in [TestConsulSecretBackend()](https://github.com/hashicorp/terraform-provider-vault/blob/cd2edae65dbb86d479c03c91164512e2863fdf54/vault/resource_consul_secret_backend_test.go#L13-L14). Even though these parameters don't come back from vault, the terraform state can be checked in tests.

**Comment by mr-miles** [comment]
Good idea, I added some tests for the extra properties

**Review by tvoran — APPROVED** [review]
Looks good, thanks!

**Pull request merged**